src/app/common/context/MustGatherContext.tsx

```diff
@@ -0,0 +1,118 @@
+import * as React from 'react';
+import { useMustGathersQuery } from '@app/queries';
+import { UseQueryResult } from 'react-query';
+import { IMustGatherResponse, mustGatherStatus } from '@app/client/types';
+import { MustGatherWatcher } from '@app/common/components/MustGatherWatcher';
+import { AppContext } from '@app/common/context/AppContext';
+
+export type mustGatherObjType = {
+  customName: string;
+  type: 'plan' | 'vm';
+  status: mustGatherStatus;
+};
+
+export type mgWatchListType = {
+  name: string;
+  isGathering: boolean;
+}[];
+
+export type mustGatherListType = mustGatherObjType[];
+interface IMustGatherContext {
+  mustGatherModalOpen: boolean;
+  setMustGatherModalOpen: (isOpen: boolean) => void;
+  mustGatherList: mustGatherListType;
+  setActiveMustGather: (mustGatherObj: mustGatherObjType) => void;
+  activeMustGather: mustGatherObjType | undefined;
+  mustGatherWatchList: mgWatchListType;
+  setMustGatherWatchList: (list: mgWatchListType) => void;
+  mustGathers: UseQueryResult<IMustGatherResponse[], Response>;
+}
+
+const mustGatherContextDefaultValue = {} as IMustGatherContext;
+export const MustGatherContext = React.createContext<IMustGatherContext>(
+  mustGatherContextDefaultValue
+);
+
+interface IMustGatherContextProvider {
+  children: React.ReactNode;
+}
+
+export const MustGatherContextProvider: React.FunctionComponent<IMustGatherContextProvider> = ({
+  children,
+}: IMustGatherContextProvider) => {
+  const appContext = React.useContext(AppContext);
```

Comment:
Hm... okay, AppContext does make sense as a name if it means it's the only context we guarantee all other context providers should have access to. Hmm.

Reply:
this was part of the thought process, but I suppose it's ok for Providers to depend on other arbitrary providers - we just have to keep the component nesting in order